### PR TITLE
Waiting Room [ #4 ]( https://github.com/witseie-elen4010/2024-group-lab-003/issues/4 )

### LINKED ISSUE #4: As a player I can view the players in the waiting room of the game

### PR DESCRIPTION
**Pull request by Yishai-Bass**
closes #4 

# Waiting Room

## Description 

Created a basic waiting room screen wherein the player can see all the players' nicknames who are in the room in real time. The players view a loading icon telling them to wait for the admin to start the game

## Acceptance Criteria

1. The user can enter the waiting room from having created a game
2. The user can enter the waiting room from having joined a game using a game code
3. The player is able to view the game code at the top of the screen
4. The player is able to view the current room players on the side
5. The room members update in real time should a new player join the room

## Story Points
*4* Story points

### CODE REVIEW

**Review by JosephKaplan613 — CHANGES_REQUESTED** [review]
Design: Approved
Functionality: Rejected. Leave button not functional and out of scope
Complexity: Approved
Tests: Approved
Naming: Approved
Comments: Approved
Style: Approved
Documentation: Approved

File: src/views/waitingRoom.html [hunk 1]
```diff
@@ -1 +1,49 @@
...
+    <link rel="stylesheet" href="/css/waitingRoom.css">
+</head>
+<body>
+    <div class="container-fluid full-height">
+        <div class="row full-height">
+            <!-- Sidebar for players list -->
+            <div class="col-md-3 col-sm-12 flex-column-between">
+                <div>
+                    <!-- Leave button -->
+                    <button id="leaveButton" type="button" class="btn btn-outline-secondary">Leave</button>
```
[the remark this answers is not in the document]
Comment (JosephKaplan613) [reply src/views/waitingRoom.html]: Well done. Now ready for merge with granular user story.

**Review by JosephKaplan613 — COMMENTED** [review]
Design: Approved
Functionality: Approved
Complexity: Approved
Tests: Approved
Naming: Approved
Comments: Approved
Style: Approved
Documentation: Approved

**Review by JosephKaplan613 — COMMENTED** [review]
Design: Approved
Functionality: Approved
Complexity: Approved
Tests: Approved
Naming: Approved
Comments: Approved
Style: Approved
Documentation: Approved

**Review by akivalevitt — COMMENTED** [review]
Design: Approved
The design effectively integrates real-time updates and user interaction, providing a waiting room experience where players can see updates dynamically.

Functionality: Approved 
The functionality comprehensively addresses all aspects of the user story. Players can seamlessly enter the waiting room after creating or joining a game, view the game code at the top of the screen, and see a list of current room players that updates in real time on the side of the screen. 

Complexity: Approved
The complexity of the real-time update feature is well-managed, suggesting the use of efficient data handling and state management practices.

Tests: Approved

Naming: Approved
Naming conventions within the code are clear and intuitive, facilitating easy navigation and maintenance of the codebase

Comments: Approved
Comments within the code provide valuable insights into the logic and design choices, particularly regarding how real-time updates are handled and displayed.

Style: Approved
The code strictly adheres to common JavaScript conventions, such as using const for variables that are not re-assigned, and arrow functions for event handlers. 

Documentation: Approved

**Pull request merged**